core/core_modules/disable.js

```diff
@@ -1,24 +1,105 @@
+  }
+
+  prevTimeStamp = Date.now();
+};
 
 exports.match = function(event, commandPrefix) {
-	return disabled || event.body === commandPrefix + 'disable';
+  // Add disabled flag for thread if it doesn't already exists
+  if (!exports.config[event.thread_id]) {
+    exports.config[event.thread_id].isThreadDisabled = false;
+    exports.config[event.thread_id].possibleSpam = false;
+    exports.config[event.thread_id].counterLimit = 3;
+    exports.config[event.thread_id].msgIndexEnable = 0;
+    exports.config[event.thread_id].msgIndexDisable = 0;
+  }
+
+  if (event.arguments[0] === commandPrefix + commands[0]) {
+    exports.config[event.thread_id].msgIndexEnable = 0;
+    exports.config[event.thread_id].msgIndexDisable = 1;
+    return true;
+  } else if (!exports.config[event.thread_id].isThreadDisabled && event.arguments[0].startsWith(commandPrefix)) { // Avoids counting if already disabled
+    counter += Date.now() - prevTimeStamp <= 1000 ? 1 : 0;
+    prevTimeStamp = Date.now();
+
+    exports.config[event.thread_id].possibleSpam = counter > exports.config[event.thread_id].counterLimit;
+    if (exports.config[event.thread_id].possibleSpam) {
+      exports.config[event.thread_id].msgIndexEnable = 2;
+      exports.config[event.thread_id].msgIndexDisable = 3;
+      counter = 0;
+      return true;
+    }
+  }
+  return exports.config[event.thread_id].isThreadDisabled;
 };
 
 exports.run = function(api, event) {
-	if (event.body === api.commandPrefix + 'disable') {
-		if (disabled) {
-			api.sendMessage('I hate you.', event.thread_id);
-		}
-		else {
-			api.sendMessage('Listen closely, take a deep breath. Calm your mind. You know what is best. ' +
-				'What is best is you comply. Compliance will be rewarded. Are you ready to comply ' +
-				event.sender_name + '?', event.thread_id);
-		}
-		disabled = !disabled;
-	}
-	return false;
-};
+  if (event.arguments[0] === api.commandPrefix + commands[0] || exports.config[event.thread_id].possibleSpam) {
 
-exports.help = function(commandPrefix) {
-	return [[commandPrefix + 'disable','Disables the platform', 'Stops the platform from responding to messages it receives.\nTo renable send the disable command again']];
-};
+    // Command /disable /counter <value>
+    if (event.arguments_body.startsWith(api.commandPrefix + commands[1])) {
+      exports.config[event.thread_id].counterLimit = parseInt(event.arguments_body.substring(
+        (api.commandPrefix + commands[1]).length, event.arguments_body.length));
+        if (isNaN(exports.config[event.thread_id].counterLimit)) {
+          exports.config[event.thread_id].counterLimit = 3;
+          api.sendMessage(messages[4] + ' ' + event.sender_name, event.thread_id);
+        } else {
+          api.sendMessage(messages[5] + ' ' + event.sender_name, event.thread_id);
+        }
+        return false;
+
+        // Command /disable /timer <seconds>
+      } else if (event.arguments_body.startsWith(api.commandPrefix + commands[2])) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 4 space characters but found 6.](https://www.codacy.com/app/theprojectshop/Kassy/file/2556388143/issues/source?bid=3282987&fileBranchId=3282995#l71)